src/hex.ts

```diff
@@ -55,6 +65,50 @@ export function assertIsStrictHexString(value: unknown): asserts value is Hex {
   );
 }
 
+/**
+ * Validate that the passed prefixed hex string is a valid hex address, or a
+ * valid mixed-case checksum address.
+ *
+ * @param possibleAddress - Input parameter to check against.
+ * @returns Whether or not the input is a valid hex address.
+ */
+export function isValidHexAddress(possibleAddress: Hex) {
+  return (
+    is(possibleAddress, HexAddressStruct) ||
+    isValidChecksumAddress(possibleAddress)
+  );
+}
+
+/**
+ * Validate that the passed hex string is a valid ERC-55 mixed-case
+ * checksum address.
+ *
+ * @param possibleChecksum - The hex address to check.
+ * @returns True if the address is a checksum address.
+ */
+export function isValidChecksumAddress(possibleChecksum: Hex) {
+  if (!is(possibleChecksum, HexChecksumAddressStruct)) {
+    return false;
+  }
+
+  const unPrefixed = remove0x(possibleChecksum);
+  const unPrefixedHash = remove0x(
+    bytesToHex(keccak256(unPrefixed.toLowerCase())),
+  );
+
+  for (let i = 0; i < unPrefixedHash.length; i++) {
```

Comment:
Is the hash guaranteed to be the same length as the address? I noticed that the examples in the EIP seem to loop over the address characters rather than the hash.